pyiceberg/expressions/__init__.py

```diff
@@ -559,12 +561,19 @@ def as_bound(self) -> Type[BoundNotNaN[L]]:
         return BoundNotNaN[L]
 
 
-class SetPredicate(UnboundPredicate[L], ABC):
-    literals: Set[Literal[L]]
+class SetPredicate(UnboundPredicate[L], IcebergBaseModel, ABC):
+    type: str = Field(default="in", alias="type")
```

Comment:
I think we can use [`Literal`](https://peps.python.org/pep-0586/) here:
```suggestion
    type: Literal["in"] = Field(default="in", alias="type")
```
The only acceptable value is `in` for this predicate